features/org.wso2.carbon.appmgt.store.feature/src/main/resources/admin-dashboard/site/themes/default/templates/businessowners-task/js/businessowners.js

```diff
@@ -0,0 +1,264 @@
+    jQuery('html, body').animate({
+                                     scrollTop: section.offset().top
+                                 }, 1000);
+    ;
+}
+var businessOwnersArray = new Array(); // xacml policy details array
+var editedOwnerId = 0; //if 1 then edit else save
+var context = this.jagg.site.context;
+var tags = [];
+var editor;
+var extraFieldCount = 0;
+
+function completeAfter(cm, pred) {
+    var cur = cm.getCursor();
+    if (!pred || pred()) {
+        setTimeout(function () {
+            if (!cm.state.completionActive) {
+                cm.showHint({completeSingle: false});
+            }
+        }, 100);
+    }
+    return CodeMirror.Pass;
+}
+
+function completeIfAfterLt(cm) {
+    return completeAfter(cm, function () {
+        var cur = cm.getCursor();
+        return cm.getRange(CodeMirror.Pos(cur.line, cur.ch - 1), cur) == "<";
+    });
+}
+
+function completeIfInTag(cm) {
+    return completeAfter(cm, function () {
+        var tok = cm.getTokenAt(cm.getCursor());
+        if (tok.type == "string" && (!/['"]/.test(tok.string.charAt(tok.string.length - 1)) || tok.string.length
+                                                                                               == 1)) {
+            return false;
+        }
+        var inner = CodeMirror.innerMode(cm.getMode(), tok.state).state;
+        return inner.tagName;
+    });
+}
+
+
+$(document).ready(function () {
+    // Get shared partials
+    $.ajax({
+               url: context + '/apis/businessowners/list',
+               type: 'GET',
+               contentType: 'application/json',
+               dataType: 'json',
+               success: function (data) {
+                   for (var i = 0; i < data.length; i++) {
+                       businessOwnersArray.push({
+                                                    owner_id: data[i].owner_id,
+                                                    owner_name: data[i].owner_name,
+                                                    owner_mail: data[i].owner_email,
+                                                    owner_desc: data[i].owner_desc,
+                                                    owner_site: data[i].owner_site,
+                                                    keys: data[i].keys,
+                                                    values: data[i].values
+                                                });
+                   }
+                   updateOwners();
+               },
+               error: function () {
+               }
+           });
+    $('#policy-name').select();
+});
+
+//save event
+$(document).on("click", "#btn-owner-save", function () {
+
+    var ownerName = $('#owner-name').val();
+    var ownerMail = $('#owner-email').val();
+    var description = $('#owner-desc').val();
+    var siteLink = $('#owner-site').val();
+    var keys = "";
+    var values = "";
+
+    if (extraFieldCount > 0) {
+        var i = extraFieldCount;
+        while (i > 1) {
```

Comment:
As mentioned above comments, use a Map to maintain extra details of a business owner